lib/core/test/unit/Cardano/Pool/DB/Properties.hs

```diff
@@ -1284,6 +1294,38 @@ prop_MultiPoolCertificateSequence_coverage mpcs = checkCoverage
     certificateSequences = getSinglePoolSequences mpcs
     poolCount = length certificateSequences
 
+
+-- | read . put == pure
+prop_putHeaderListHeader
+    :: DBLayer IO
+    -> [BlockHeader]
+    -> Positive Int
+    -> Property
+prop_putHeaderListHeader DBLayer{..} headers (Positive k) =
+    monadicIO (setup >> prop)
+  where
+    sortedHeaders = sortBy (\a b -> compare (blockHeight a) (blockHeight b))
+        . L.nubBy (\a b -> (blockHeight a) == (blockHeight b)) . nub $ headers
+    expected = reverse . (if k == 0 then id else take k) . reverse $ sortedHeaders
```

Comment:
This simulates the limiting of the results to the latest k elements, while still retaining order of oldest block first.